lib/p2p/Peer.ts

```diff
@@ -94,6 +119,26 @@ class Peer extends EventEmitter {
       delete this.socket;
     }
 
+    if (this.pingTimer) {
+      clearInterval(this.pingTimer);
+      delete this.pingTimer;
+    }
+
+    if (this.stallTimer) {
+      clearInterval(this.stallTimer);
+      delete this.stallTimer;
+    }
+
+    if (this.connectTimeout) {
+      clearTimeout(this.connectTimeout);
+      delete this.connectTimeout;
```

Comment:
`clearTimeout` will remove the timeout from the scheduler, but the id will remain on the variable. I think it's good to clean it, to indicate that there's no active scheduler. otherwise you cannot know that.
I'm actually gonna change this to `null` instead of `undefined`. I think it's more appropriate when to trying represent "empty" state.